crossing-repair.lic

```diff
@@ -2,51 +2,199 @@
+custom_require.call(%w[common common-money common-travel equipmanager])
 
 class CrossingRepair
-  include DRC
-  include DRCM
-  include DRCT
 
   def initialize
-    settings = setup
+    arg_definitions = [
+      [
+        { name: 'town', regex: $HOMETOWN_REGEX, optional: true, description: 'Town to repair in' },
+        { name: 'force', regex: /force/i, optional: true, description: 'Force repair gear, ignoring any wait timers' },
+        { name: 'debug', regex: /debug/i, optional: true, description: 'Enable debug mode' }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+    $debug_mode_crossing_repair = args.debug || UserVars.crossing_repair_debug
+    town = DRC.get_town_name(args.town)
+    settings = get_settings
+
+    # If town argument is not recognized, choosing to exit the script with error
+    # instead of continue with the character's hometown because if the player
+    # wanted to repair in a specific town then they might be far away from their
+    # hometown and not want the script to suddenly rush them off,
+    # especially if it was a simple typo or the player suffers social outrage.
+    if args.town && !town
+      DRC.message("Could not identify town for argument: #{args.town}")
+      DRC.message("To avoid ambiguity, please use the town's full name: https://elanthipedia.play.net/Category:Cities")
+      exit
+    end
+
+    # If the user has specified a repair town via script call argument, set that here.
+    # This argument has already passed through the $HOMETOWN_REGEX check so we know it to be
+    # a valid town.
+    settings['hometown'] = town if town
+
+    # Check if we have the repair_heuristic_gear yaml settings. This uses appraise to assess
+    # condition of listed items, and if damaged to a certain thresshold, then run crossing-repair
+    # on _all_ items. In other words, try to repair all your gear if some key items show wear and tear.
+    @repair_heuristic_gear = settings.repair_heuristic_gear
+    @repair_heuristic_threshold = settings.repair_heuristic_threshold
+
+    # Set up a repair timer so we can use this metric to know whether to repair.
+    # User has option via script arg to force repair and ignore any repair timers.
+    # Also handle that here.
+    @repair_timer = settings.repair_timer
+    UserVars.repair_timer_snap ||= Time.now
+    unless args.force || should_repair?
+      DRC.message("Skipping repair") if $debug_mode_crossing_repair
+      exit
+    end
+
+    # Set up equipment manager. Wear our standard equipment set. Empty our hands.
+    @equipment_manager = EquipmentManager.new
+    @equipment_manager.wear_equipment_set?('standard')
+    @equipment_manager.empty_hands
 
+    # Handle some bank and money items. Set hometown data.
+    @skip_bank = settings.sell_loot_skip_bank
+    amount, denom = settings.sell_loot_money_on_hand.split(' ')
+    @keep_copper = DRCM.convert_to_copper(amount, denom)
+    @repair_withdrawal_amount = settings.repair_withdrawal_amount
```

Comment:
A number of these instance vars are pulled from settings, but we pass all the settings into `repair_handler` and `repair_prep` anyway. What do you think about removing those kinds of instance vars and pulling them from `settings`?